### PR TITLE
feat: add oraclePriceAndTimestamp to the market data

### PR DESCRIPTION
**Pull request by evgenyboxer**

### CODE REVIEW

File: contracts/BinaryOptionMarketData.sol [hunk 1]
```diff
@@ -108,6 +108,11 @@ interface IBinaryOptionMarket {
         uint finalPrice;
     }
 
+    struct OraclePriceAndTimestamp {
```
Comment (zyzek) [inline contracts/BinaryOptionMarketData.sol]: This struct doesn't actually exist in the `BinaryOptionMarket`, so it should actually be declared in the `BinaryOptionMarketData` contract.

File: contracts/BinaryOptionMarketData.sol [hunk 2]
```diff
@@ -189,6 +194,7 @@ contract BinaryOptionMarketData {
         address creator;
         IBinaryOptionMarket.Options options;
         IBinaryOptionMarket.Times times;
+        IBinaryOptionMarket.OraclePriceAndTimestamp oraclePriceAndTimestamp;
```
Comment (zyzek) [inline contracts/BinaryOptionMarketData.sol]: This `MarketParameters` struct is for things which don't change over the lifetime of the contract, while the `MarketData` one is for those that do change. The oracle price changes, so it belongs in `MarketData`.

It's probably worth adding comments on this just above the structs to make this clear for others reading the code as well.

File: contracts/BinaryOptionMarketData.sol [hunk 3]
```diff
@@ -215,13 +221,15 @@ contract BinaryOptionMarketData {
 
         (IBinaryOption long, IBinaryOption short) = market.options();
         (uint biddingEndDate, uint maturityDate, uint expiryDate) = market.times();
+        (uint price, uint updatedAt) = market.oraclePriceAndTimestamp();
```
Comment (zyzek) [inline contracts/BinaryOptionMarketData.sol]: This is causing a stack too deep error. Moving it to the `getMarketData` function would solve the issue.

**Review by zyzek — CHANGES_REQUESTED** [review]

**Comment by evgenyboxer** [comment]
@zyzek all fixed now.

**Review by zyzek — APPROVED** [review]

**Comment by zyzek** [comment]
Looks good to me.

**Pull request merged**